pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -115,19 +115,23 @@ func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec,
 		if err != nil {
 			return spec, changed, fmt.Errorf("failed to get vsphere client: '%v'", err)
 		}
-		defer client.Logout(ctx)
+		defer func() {
+			if lerr := client.Logout(ctx); lerr != nil {
+				glog.Errorf("vsphere client failed to logout: %s", lerr)
```

Comment:
Instead of logging those situations, we should call `utilruntime.HandleErr`
It also logs the error, but additionally increases the error-counter-metric

Reply:
Fixed